linker/descriptors.go

```diff
@@ -326,31 +342,31 @@ func (r *result) storeOptionBytesInField(fld, origFld *descriptorpb.FieldDescrip
 
 type fileImports struct {
 	protoreflect.FileImports
-	parent *result
+	files []protoreflect.FileImport
+}
+
+func (r *result) createImports() fileImports {
+	fd := r.FileDescriptorProto()
+	imps := make([]protoreflect.FileImport, len(fd.Dependency))
+	for i, dep := range fd.Dependency {
+		desc := r.deps.FindFileByPath(dep)
+		imps[i] = protoreflect.FileImport{FileDescriptor: desc}
+	}
+	for _, publicIndex := range fd.PublicDependency {
+		imps[int(publicIndex)].IsPublic = true
+	}
+	for _, weakIndex := range fd.WeakDependency {
+		imps[int(weakIndex)].IsPublic = true
```

Comment:
Shouldn't this be `IsWeak = true`?

Reply:
YES! Good catch. I'm glad you have eagle eyes. Some of the characteristics of these descriptor impls are clearly not fully tested.